src/analyzer/engines/engine.py

```diff
@@ -0,0 +1,26 @@
+class ModelingEngine():
+    """
+    ModelingEngine defines a common interface for performing text modeling across
+    different modeling APIs.
+    """
+
+    def __init__(self):
+        self.models = {}
+
+    def fit(self, model, dictionary, corpus):
+        """
+        Fits a set of models on the provided corpus.
+        """
+        raise NotImplementedError("fit() must be implemented by a subclass.")
```

Comment:
We might also consider specifying that `fit()` methods return `self`, which *should* make them compatible with the [sklearn API](https://scikit-learn.org/stable/developers/develop.html#fitting) (meaning we could theoretically put them into a sklearn `Pipeline`). If we want to go this route, we'll also want to create a `predict` and/or `transform` methods &mdash; possibly `predict` would just call `topics` and `transform` could just call `update`?